src/components/core/GeoChart.tsx

```diff
@@ -69,6 +83,31 @@ export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
         };
         return legendRenderer(legendProps);
     };
+
+    private validateData = (): void => {
+        const {
+            config: { limit, mdObject: { buckets = [] } = {} },
+            execution,
+            onDataTooLarge,
+        } = this.props;
+
+        if (!execution) {
+            return;
+        }
+
+        const geoDataIndex: IGeoDataIndex = getGeoDataIndex(buckets);
+        if (
+            isDataOfReasonableSize(
+                execution.executionResult,
+                geoDataIndex,
+                limit || DEFAULT_DATA_POINTS_LIMIT,
```

Comment:
`limit` default prop should be `DEFAULT_DATA_POINTS_LIMIT `, right?

Reply:
updated